content/en/docs/concepts/services-networking/service-topology.md

```diff
@@ -15,7 +15,15 @@ weight: 10
 
 <!-- overview -->
 
-{{< feature-state for_k8s_version="v1.17" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.21" state="deprecated" >}}
+
+{{< note >}}
+
+Service Topology is deprecated as of Kubernetes v1.21, and will be removed in
+v1.22. Topology Aware Hints was introduced in Kubernetes v1.21 and provides
+similar functionality.
```

Comment:
Do we also need to update https://kubernetes.io/docs/reference/using-api/deprecation-guide/ ?

Reply:
That's a good question. I think that specific doc is focused on deprecations of API versions, not fields. I know we'll have a deprecation notice in the 1.21 release notes, but maybe we need something more visible in the docs?

/cc @liggitt

Reply:
Yeah, that page is currently focused on entirely deprecated API versions that require migration to another API version. Since v1beta1 EndpointSlices will stop being served in 1.25, and the v1 replacement is ready, they should be marked as deprecated, and added to that doc, along with a mention of the `topology` field not being writeable in v1.